products/revenue_analytics/backend/models.py

```diff
@@ -0,0 +1,203 @@
+from typing import cast, Any, Optional
+import posthoganalytics
+
+from posthog.hogql import ast
+from posthog.warehouse.models.external_data_source import ExternalDataSource
+from posthog.warehouse.models.table import DataWarehouseTable
+from posthog.models.exchange_rate.sql import EXCHANGE_RATE_DECIMAL_PRECISION
+from posthog.schema import CurrencyCode
+from posthog.hogql.database.models import (
+    BooleanDatabaseField,
+    DateTimeDatabaseField,
+    ExpressionField,
+    IntegerDatabaseField,
+    StringDatabaseField,
+    Table,
+    FieldOrTable,
+)
+from posthog.hogql.database.schema.exchange_rate import (
+    DEFAULT_CURRENCY,
+    convert_currency_call,
+)
+
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from posthog.hogql.context import HogQLContext
+
+# Stripe represents most currencies with integer amounts multiplied by 100,
+# since most currencies have its smallest unit as 1/100 of their base unit
+# It just so happens that some currencies don't have that concept, so they're listed here
+# https://docs.stripe.com/currencies#zero-decimal
+ZERO_DECIMAL_CURRENCIES_IN_STRIPE: list[CurrencyCode] = [
+    CurrencyCode.BIF,
+    CurrencyCode.CLP,
+    CurrencyCode.DJF,
+    CurrencyCode.GNF,
+    CurrencyCode.JPY,
+    CurrencyCode.KMF,
+    CurrencyCode.KRW,
+    CurrencyCode.MGA,
+    CurrencyCode.PYG,
+    CurrencyCode.RWF,
+    CurrencyCode.UGX,
+    CurrencyCode.VND,
+    CurrencyCode.VUV,
+    CurrencyCode.XAF,
+    CurrencyCode.XOF,
+    CurrencyCode.XPF,
+]
+
+BASE_FIELDS: dict[str, FieldOrTable] = {
```

Comment:
This + some lines below is where we're creating the view. This will eventually become a materialized view that's automatically created by the system, but this can live like a virtual view for now.